src/commands/push-image.yml

```diff
@@ -14,11 +14,16 @@ parameters:
     description: Name of an Amazon ECR repository
 
   tag:
-    description: A docker image tag (default = latest)
+    description: Optionally define a single docker image tag to push, instead of pushing all the repo's tags (default = "")
     type: string
-    default: "latest"
+    default: ""
 
 steps:
   - run:
       name: Push image to Amazon ECR
-      command: "docker push $<<parameters.account-url>>/<<parameters.repo>>:<<parameters.tag>>"
+      command: |
+        tag_arg=""
+        if [ "<<parameters.tag>>" != "" ]; then
+          tag_arg=":<<parameters.tag>>"
+        fi
+        docker push $<<parameters.account-url>>/<<parameters.repo>>${tag_arg}
```

Comment:
This is awesome, thank you!

Reply:
It'd be good if the push-image command could also support a comma-separated list of tags like the build-image command, so that they can work well together when used by the `build-and-push-image` command and job (which internally invokes `build-image` and `push-image`, supplying the same `tag` parameter to both)

Reply:
thank you for the review @lokst . I now remember why I've gone for pushing all tags as default - `docker push` doesn't support pushing multiple tags in a single invocation - it does however support pushing all tags of an image at once. I think the best I can do is multiple invocations of `docker push` in a loop, I suppose it's not so bad is it?